### PR TITLE
Add a few tests

### PR DESCRIPTION
**Pull request by pierreablin**
This adds some tests in the main function, to check that the input are of the correct shape, and are symmetric + positive matrices

### CODE REVIEW

File: qndiag/qndiag.py [hunk 1]
```diff
@@ -74,6 +78,20 @@ def qndiag(C, B0=None, weights=None, max_iter=1000, tol=1e-6,
     https://arxiv.org/abs/1811.11433
     """
     t0 = time()
+    if not isinstance(C, np.ndarray):
+        raise TypeError('Input tensor C should be a numpy array.')
+    if C.ndim != 3:
+        raise ValueError('Input tensor C should have 3 dimensions')
```
Comment (agramfort) [inline qndiag/qndiag.py]:
```suggestion
        raise ValueError(f'Input tensor C should have 3 dimensions (Got {C.ndim})')
```

File: qndiag/qndiag.py [hunk 2]
```diff
@@ -74,6 +78,20 @@ def qndiag(C, B0=None, weights=None, max_iter=1000, tol=1e-6,
...
+    if C.shape[1] != C.shape[2]:
+        raise ValueError('The last two dimensions of C should be the same')
+    if check_sympos:
+        if not np.allclose(C, C.swapaxes(-1, -2)):
+            raise ValueError('C does not contain only symmetric matrices')
+        try:  # Check positivity using Cholesky
+            np.linalg.cholesky(C)
+        except np.linalg.LinAlgError:
+            raise ValueError('C contains some non positive matrices') from None
```
Comment (agramfort) [inline qndiag/qndiag.py]: why the `from None` ?

File: qndiag/qndiag.py [hunk 3]
```diff
@@ -143,6 +161,30 @@ def qndiag(C, B0=None, weights=None, max_iter=1000, tol=1e-6,
...
+        The set of covariance matrices
+
+    diag_only : bool, optional
+        Whether to return the diagonal of the dataset only
+
+    Returns
+    -------
+    op : array-like
+        Array of shape (n_samples, n_features, n_features)
+        if diag_only is False, else (n_samples, n_features, n_features)
```
Comment (agramfort) [inline qndiag/qndiag.py]:
```suggestion
        if diag_only is False, else (n_samples, n_features)
```

File: qndiag/qndiag.py (hunk 2 above)
Comment (pierreablin) [reply qndiag/qndiag.py]: weird hack found here: https://stackoverflow.com/questions/52725278/during-handling-of-the-above-exception-another-exception-occurred

without it, it prints two exceptions, first the LinAlgError and then the ValueError

**Pull request merged**